app/controllers/madmin/resource_controller.rb

```diff
@@ -91,13 +98,27 @@ def new_resource_params
     def change_polymorphic(data)
       return data unless data.is_a?(ActionController::Parameters) && data[:type]
 
-      if data[:type] == "polymorphic"
-        GlobalID::Locator.locate(data[:value])
-      else
-        raise "Unrecognised param data: #{data.inspect}"
+      raise "Unrecognised param data: #{data.inspect}" unless data[:type] == "polymorphic"
+
+      GlobalID::Locator.locate(data[:value])
+    end
+
+    def parse_json_attribute(key, value)
+      return value unless json_attribute?(key)
+      return value unless value.is_a?(String) && value.present?
```

Comment:
Since this is submitted from a form, it'll always be a String, so I don't know that we need to check this?